### PR TITLE
Fix for piped streams

### PR DESCRIPTION
**Pull request by masbug**
Hi,

this patch is a fix for piped streams on Linux (I found no issue on Windows using the old code).
It wraps pipes using a stream pump with minimal buffer.
Ref issue: https://github.com/spatie/dropbox-api/issues/37

Best regards

### CODE REVIEW

**Comment by masbug** [comment]
*ping* @freekmurze  :)

File: src/Client.php [hunk 1]
```diff
@@ -371,7 +371,19 @@ public function uploadChunked(string $path, $contents, $mode = 'add', $chunkSize
...
+            /** @var resource $contents */
+            $stream = new \GuzzleHttp\Psr7\PumpStream(function ($length) use ($contents) {
+                $data = fread($contents, $length);
+                if (strlen($data) === 0) {
+                    return false;
+                }
+
+                return $data;
+            });
+        } else {
```
Comment (freekmurze) [inline src/Client.php]: Is this else statement needed?

Comment (masbug) [reply src/Client.php]: Yes, it's a speedup for non-piped contents...which I believe is the majority. :)

Comment (freekmurze) [reply src/Client.php]: But won't the results be the same if the `else` is removed? There's a `return` statement for each path inside the `if` above.

Comment (masbug) [reply src/Client.php]: This is the whole chunk:
`
        if ($this->isPipe($contents)) {
            /** @var resource $contents */
            $stream = new \GuzzleHttp\Psr7\PumpStream(function ($length) use ($contents) {
                $data = fread($contents, $length);
                if (strlen($data) === 0) {
                    return false;
                }
                return $data;
            });
        } else {
            $stream = Psr7\stream_for($contents);
        }`

Return statement is inside a closure...so it doesn't affect the outer `if`.

The point of these changes is to initialize `$stream` differently depending on whether input stream is piped (in which case it has to be pumped) or not (this is where we can speed it up and use less resources to process it).

**Comment by masbug** [comment]
This pull resolves issue #37 and shouldn't be closed...

**Comment by masbug** [comment]
@freekmurze Is there a special reason why this fix is not merged?

**Comment by freekmurze** [comment]
Thanks for your work on this! I was waiting on that else to be refactored away. I'll quickly do it myself.

**Pull request merged**